src/classes/PS_IntegrationService.cls

```diff
@@ -35,16 +35,108 @@
 */
 global class PS_IntegrationService {
-    
+
+    @TestVisible
+    private static final String INVALID_CONFIG_PARAMS = System.Label.gePS_IntegrationTestInvalidConfigParams;
+    private static final String ELEVATE_ONBOARDING_ERROR = System.Label.commonElevateFailedOnboarding;
+    public static final String SERVICE_CONST = 'service';
+    public static final String IS_SECRET_CONST = 'issecret';
+    public static final String KEY_CONST = 'key';
+    public static final String VALUE_CONST = 'value';
+    public static final String TRUE_CONST = 'true';
+    public static final String FALSE_CONST = 'false';
+    public static final String DELIMETER_CONST = '::';
+
+    private static final Map<String, String> fieldMappingConstants = new Map<String, String>{
+            KEY_CONST => 'Key__c',
+            VALUE_CONST => 'Value__c',
+            IS_SECRET_CONST => 'Is_Secret__c',
+            SERVICE_CONST => 'Service__c'
+    };
+
+
     /*******************************************************************************************************
     * @description Provides an endpoint for the Payment Services Integrations package
     * to call when a Customer registers, and pass parameters containing the secrets
     * needed for calls to Payment Services endpoints(Tokenize, Purchase) and creating JWTs
-    * @param secrets contains secrets, such as the api key for creating a JWT
+    * @param configParams contains secrets, such as the api key for creating a JWT
     */
     global static void setConfiguration(Map<String, List<Map<String, String>>> configParams) {
-        // TODO This method needs to be implemented after we decide on the storage mechanism
-        // for the onboarding data
 
+        if (configParams == null || configParams.isEmpty()) {
+            throw new IntegrationServiceException(INVALID_CONFIG_PARAMS);
+        }
+
+        List<Payment_Services_Configuration__c> configurations = new List<Payment_Services_Configuration__c>();
+
+        for (String service : configParams.keySet()) {
+            for (Map<String, String> valueMap : configParams.get(service)) {
+                configurations.add(createConfiguration(service, valueMap));
+            }
+        }
+
+        try {
+            upsert configurations Service_Key__c;
+        } catch (Exception ex) {
+            throw new IntegrationServiceException(ELEVATE_ONBOARDING_ERROR + ex.getMessage());
+        }
+
+    }
+
+    /******************************************************************************************************
+    * @description This method builds the a configuration object based on key value pair configuration
+    * parameters and the configuration service type
+    * @param service Configuration service type, ex: payments, makana
+    * @param keyValuePairs
+    *
+    * @return Payment Services Configuration object
+    */
+    public static Payment_Services_Configuration__c createConfiguration(
```

Comment:
Should this be private instead of public? Do we expect to call it outside of this class?